runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -72,11 +101,22 @@ protected void shutDown() throws Exception
 		removeKingdomInfobox();
 	}
 
+	@Provides
+	KingdomConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(KingdomConfig.class);
+	}
+
 	@Subscribe
 	public void onVarbitChanged(VarbitChanged event)
 	{
-		favor = client.getVar(Varbits.KINGDOM_FAVOR);
-		coffer = client.getVar(Varbits.KINGDOM_COFFER);
+		if (client.getGameState() == GameState.LOGGED_IN && isThroneOfMiscellaniaCompleted() && isInKingdom()
+			&& (getCoffer() != client.getVar(Varbits.KINGDOM_COFFER) || getFavor() != client.getVar(Varbits.KINGDOM_FAVOR)))
```

Comment:
One annoying thing I found about this code is that this will not automatically update the values when visiting Etceteria until some varbit change happens after you're there. For instance, if you travel to Etceteria via spirit tree then leave right away, these conditions will never be met because `isInKingdom()` is `false` during the varbit changes while you're changing game states.

Because of the `isKingdom()` false negative, I propose this subscriber be changed somewhat to handle this case.

* If `inKingdom()` is false and either coffer or favor values are 0, don't set any new values (this will be the case in some varbit changed events as the indexes are transmitted)
* If `inKingdom()` is false, coffer and favor are both non-zero, **and** either of their values are different from the stored values, set lastChanged and the values (the server seems to send the last known values from visiting misc/etc--if we didn't get a varbitchanged on the island itself, we should update the values from varbit changes elsewhere in the world)
* If `inKingdom()` is true and either value is different from the stored values, set lastChanged and the values

ie.

```java
public void onVarbitChanged(VarbitChanged event)
{
	final int coffer = client.getVar(Varbits.KINGDOM_COFFER);
	final int favor = client.getVar(Varbits.KINGDOM_FAVOR);

	if (client.getGameState() == GameState.LOGGED_IN
		&& isThroneOfMiscellaniaCompleted()
		&& (isInKingdom() || coffer != 0 && favor != 0)
		&& (coffer != getCoffer() && favor != getFavor()))
	{
		setLastChanged(Instant.now());
		setCoffer(coffer);
		setFavor(favor);
	}

	processInfobox();
}
```